crates/rpc-types-eth/src/transaction/mod.rs

```diff
@@ -151,7 +151,11 @@ where
     }
 
     /// Converts a consensus `tx` with an additional context `tx_info` into an RPC [`Transaction`].
-    pub fn from_transaction(tx: Recovered<T>, tx_info: TransactionInfo) -> Self {
+    pub fn from_transaction<TxIn>(tx: Recovered<TxIn>, tx_info: TransactionInfo) -> Self
```

Comment:
does this require a type hint for `Transactio::from_transaction`?

if not then we can do this without causing problems, but we should check.

Reply:
Yes, that is true unfortunately. The type must be known at some point and AFAIK cannot be set that `T = TxIn` by default.

So I guess that would cause problems.